src/Utilities.cpp

```diff
@@ -707,3 +707,18 @@ OsmAnd::PointD OsmAnd::Utilities::getTileEllipsoidNumberAndOffsetY(int zoom, dou
     res.y = floor(((xTilesCountForThisZoom / 2 - M2 * xTilesCountForThisZoom / 2 / M_PI) - yTileNumber) * tileSize);
     return res;
 }
+
+std::pair<int, int> OsmAnd::Utilities::calculateFinalXYFromBaseAndPrecisionXY(int bazeZoom, int finalZoom, int precisionXY,
+                                                                              int xBase, int yBase, bool ignoreNotEnoughPrecision) {
+    int finalX = xBase;
+    int finalY = yBase;
+    for (int zoom = bazeZoom; zoom < finalZoom; zoom++) {
+        if (!ignoreNotEnoughPrecision) {
```

Comment:
The condition does not correspond to Java. + before failing you should probably output the reason for the failure.